src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -83,7 +83,14 @@ public virtual void ApplyRule(RewriteContext context)
                             throw new ArgumentOutOfRangeException($"Unexpected server variable: `{serverVariable.Type}`");
                     }
 
-                    headerDictionary.Append(name, value);
+                    if (headerDictionary.ContainsKey(name))
```

Comment:
No, TestServer doesn't support RemoteIpAddress. The best you can do is fill it in via an inline middleware:
```
app.Use((context, next) =>
{
  context.Connection.RemoteIpAddress = ...;
  return next();
});